### ISSUE TITLE
Guidance for python mode

### ISSUE THREAD

**Issue by fgregg** [vspinu/polymode#153]
Hi @vspinu, thank you so much for this great tool!

Using this [stack overflow recipe for python+latex](https://emacs.stackexchange.com/questions/20437/polymode-with-python-and-latex-mode), I've gotten mainly set up for a good mode.

The only thing lacking, is being able to smoothly send python code chunks to the interpreter. 

[python-x](https://github.com/wavexx/python-x.el) seems like it's going in the right direction, but I wonder if you have seen another setup?

**Comment by izahn** [comment]
I don't think this has anything to do with polymode, does it? Anyway, I use https://github.com/kaz-yos/eval-in-repl/

**Comment by vspinu** [comment]
I am creating a new project poly-python under which I would like to gather serveral python oriented polymodes. where does this latex environments`\begin{pycode} ...\end{pycode}` come from? Is there a processor (weaver) for such code or it's just a latex package that does the highlighting? 

Also, is this generic or python specific?

**Comment by izahn** [comment]
I believe it comes from https://github.com/gpoore/pythontex.

**Comment by vspinu** [comment]
Thanks. I looked through the docs and it's definitively worth supporting.

**Comment by yamad** [comment]
@fgregg We are working on REPL interaction for polymode over at https://github.com/polymode/polymode/pull/129 and are looking for non-R examples to work with. Would you be willing to share an example LaTeX+Python file?

**Comment by fgregg** [comment]
Sure:

```lisp
(require 'polymode)

(defcustom pm-inner/python
  (pm-hbtchunkmode "python"
                   :mode 'python-mode
                   :head-reg  "<<.*>>="
                   :tail-reg  "@")
  "python typical chunk."
  :group 'innermodes
  :type 'object)

(defcustom pm-poly/latex-python
  (pm-polymode-one "latex-python"
                   :hostmode 'pm-host/latex
                   :innermode 'pm-inner/python)
  "latex-python typical polymode."
  :group 'polymodes
  :type 'object)

(define-polymode poly-latex+python-mode pm-poly/latex-python)
(add-to-list 'auto-mode-alist '("\\.mdw$" . poly-latex+python-mode))

```

**Comment by yamad** [comment]
Thanks, do you have an example *.mdw file?

**Comment by fgregg** [comment]
```
\documentclass{article}
\title{Example}
\begin{document}


\maketitle
<< echo = False >>=
# Set up environment

import calendar

import pweave
pweave.rcParams["chunk"]["defaultoptions"].update({
    'results' : 'raw', 
    'echo': False,
    'figfmt': '.pdf',
    'width': '3.5in'
})

import pandas as pd

import numpy as np
import matplotlib.pyplot as plt
plt.style.use('ggplot')
@

\end{document}
```

**Comment by fgregg** [comment]
`mdw` is a bad file extension. this should be texw or something.

**Comment by brandonwillard** [comment]
@fgregg, I haven't used `polymode` in a while, but, when I was using it, I had similar interests and crafted a mostly work-able setup between some minor `polymode` changes [here](https://github.com/brandonwillard/polymode/commits?author=brandonwillard) and a custom Spacemacs "layer" (i.e. Emacs package-like unit) [here](https://github.com/brandonwillard/spacemacs-custom/blob/master/layers/%2Blang/noweb/packages.el).  More specifically, I used functions like [this](https://github.com/brandonwillard/spacemacs-custom/blob/master/layers/%2Blang/noweb/packages.el#L119) to send chunks from `polymode`.
